libraries/AP_HAL_ChibiOS/shared_dma.cpp

```diff
@@ -87,6 +87,9 @@ void Shared_DMA::unregister()
 // lock one stream
 bool Shared_DMA::lock_stream(uint8_t stream_id)
 {
+    if (!is_shared(stream_id)) {
```

Comment:
I'm not sure this is safe. I'm pretty sure the dshot code for instance uses the DMA lock to mediate access, so you would need to also check whether another thread has taken the lock.

Reply:
@andyp1per can you point to the code where this is done, I don't think Shared DMA is used that way in RCOutput, its only role is to allocate dma for use for dshot and bdshot, as far as I am aware dshot stuff happens entirely from the single thread. What is DMA lock used to mediate access between?